config/initializers/define-orm-models.ts

```diff
@@ -14,14 +14,14 @@ export default {
    * any internal model representation necessary.
    */
   async initialize(application: Application): Promise<void> {
-    let container = application.container;
     let models: { [modelName: string]: typeof Model } = application.container.lookupAll('model');
     let modelsGroupedByAdapter = new Map();
-    forEach(models, (ModelClass, modelName) => {
-      if (ModelClass.abstract) {
+    forEach(models, (ModelClass: typeof Model) => {
+      if (ModelClass.hasOwnProperty('abstract') && ModelClass.abstract) {
         return;
       }
-      let Adapter = container.lookup(`orm-adapter:${ modelName }`, { loose: true }) || container.lookup('orm-adapter:application');
+      // Let the model class resolve its own adapter
+      let Adapter = ModelClass.getAdapter(application.container);
```

Comment:
Personally I didn't do it from a customization perspective, more-so from a "dry up the code" approach, but will drop this change for now and can add it back in later.